src/modules/usart.rs

```diff
@@ -0,0 +1,258 @@
+use core::prelude::*;
+use core::default::Default;
+
+use collections::vec::Vec;
+use collections::string::String;
+
+use {cmu, gpio, usart};
+use gpio::Port;
+
+use self::Location::*;
+
+type PortPin = Option<(Port, u32)>;
```

Comment:
These are just used internally to make one function-declaration a little easier to read, so I don't think it's necessary to restrict the pin.
It's an option because one of the Usart's only have pins dedicated for tx/rx, while the other have four pins available.